### ISSUE TITLE
[ray] ActorPool fails with Actor method that has num_returns>1

### ISSUE THREAD

**Issue by waldroje** [ray-project/ray#13159]
### What is the problem?
Was trying to use ActorPool with example code from the documentation, but it fails for Actor methods with num_returns>1, and returns message `unhashable type: 'list'`

*Ray version and other system information (Python version, TensorFlow version, OS):*

Ray 1.0.1, Python 3.7.7, TF 2.2, RHEL 7.7

### Reproduction (REQUIRED)
Please provide a short code snippet (less than 50 lines if possible) that can be copy-pasted to reproduce the issue. The snippet should have **no external library dependencies** (i.e., use fake or mock data / environments):

```
@ray.remote
class Foo(object):

    # Any method of the actor can return multiple object refs.
    @ray.method(num_returns=2)
    def bar(self):
        return 1, 2
    
pool = ActorPool([Foo.remote() for _ in range(2)])
for _ in range(4):
    pool.submit(lambda a,v: a.bar.remote(), None)
while pool.has_next():
    result = pool.get_next(timeout=None)
    print(result)
```
If the code snippet cannot be run by itself, the issue will be closed with "needs-repro-script".

- [x] I have verified my script runs in a clean environment and reproduces the issue.
- [ ] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

**Comment by wangwwno1** [comment]
I have reproduced this issue on:  Ray 1.1.0, Python 3.8.5 PyTorch1.7.0, Ubuntu18.04
Without the wrapper `@ray.method` I can run the code smoothly
```
import ray
from ray.util import ActorPool

ray.init()
@ray.remote
class Foo(object):

    # Remove ray.method wrapper
    def bar(self):
        return 1, 2


pool = ActorPool([Foo.remote() for _ in range(2)])
for _ in range(4):
    pool.submit(lambda a, v: a.bar.remote(), None)
while pool.has_next():
    a, b = pool.get_next(timeout=None)
    print(a, b)
```
If every function can return a tuple (which is not, I guess :) there would be no need for such wrapper.
So my question is: *what is the best practice to use wrapper `ray.method`?**

**Comment by waldroje** [comment]
I actually didn't realize you could call the actor method without the decorator... what is the difference then, assuming the return was a single value and it did not error? Does it change the resources used?

**Comment by wangwwno1** [comment]
> I actually didn't realize you could call the actor method without the decorator... what is the difference then, assuming the return was a single value and it did not error? Does it change the resources used?


I'm not sure about the resource usage, but if the returned object belongs to a builtin type (or a composite of them) it should make no difference, as a python function with multiple outputs will return a tuple that contains all the values.

Take the method `Foo.bar` as an example, if I receive its outputs with `a, b = out = pool.get_next(timeout=None)`, the variable `out` would be a tuple that contains two integer (i.e **Tuple[int, int]**) and I can unpack the tuple into two variable `a` and `b`.

However, if the returned object cannot be hashed or pickled, it would raise error regardless of the existence of `ray.method`, so I'm curious about the practice of wrapper `ray.method`.

**Comment by waldroje** [comment]
My thought was as a decorated actor method, it would only utilize resources that had been assigned an actor… as opposed to a general task which would grab additional resources from the cluster….

**Comment by wangwwno1** [comment]
> My thought was as a decorated actor method, it would only utilize resources that had been assigned an actor… 

Try to specify the resource with `Actor.remote(num_cpu=2, num_gpu=0.2)`?

**Comment by micahtyong** [comment]
Hi @waldroje, I've just opened up a fix for this issue in #15402 so that `ActorPool` methods run smoothly with or without the `@ray.method` decorator. Feel free to take a look. cc: @rkooo567

**Comment by waldroje** [comment]
thanks! will take a look.
